### PR TITLE
[OIS] Add all-clusters-app Matter example

### PR DESCRIPTION
**Pull request by ATmobica**
Improvements: 
 - creating a supported apps list and using it in the build script - easier to add a new application
 - adding Mutex to guard the log message buffer in the logging module - fixing logs race condition

New features:
 - adding all-clusters-app implementation for the OIS platform

### CODE REVIEW

File: examples/all-clusters-app/openiotsdk/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+For information on how to build and run this example and further information
+about the platform it is run on see
+[Open IoT SDK examples](../../../docs/examples/openiotsdk_examples.md).
+
+The example name to use in the scripts is `all-clusters-app`.
+
+## Using the example
+
+Communication with the application goes through the active telnet session. When
+the application runs these lines should be visible:
```
Comment (pan-) [inline examples/all-clusters-app/openiotsdk/README.md:21]:
```suggestion
the application runs, these lines should be visible:
```

File: src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+LOCK_CTRL_TEST_PIN_CODE = 12345
+LOCK_CTRL_TEST_USER_INDEX = 1
+LOCK_CTRL_TEST_ENDPOINT_ID = 1
+LOCK_CTRL_TEST_USER_NAME = 'testUser'
+LOCK_CTRL_TEST_CREDENTIAL_INDEX = 1
+
+
+@pytest.mark.ctrltest
+def test_lock_ctrl(device, controller):
```
Comment (pan-) [inline src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py:78]: I believe we should test for some other clusters.

Comment (ATmobica) [reply src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py:78]: Agree, but I'm not sure which one. Not all clusters have implemented handles that allow us to perform actions useful for integration tests. I'm rather thinking about some basic clusters such as `Basic` or 'AccessControl' clusters that can be validated here. Feature-specific clusters will be validated in other examples. What do you think about it?

Comment (pan-) [reply src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py:78]: The more I think about it, the less my remark makes sense as the certification test suite should validate other clusters. 
Let's keep this as a smoke test. Details will be validated by the certification test suite.

**Review by pan- — APPROVED** [review]

**Comment by pan-** [comment]
A rebase is required @ATmobica .

File: src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    #                                [entry0, entry1, entry2, entry3])
+    # assert err == 0
+    # assert res[0].Status == chip.interaction_model.Status.Success
+
+    # err, res = read_zcl_attribute(devCtrl, "AccessControl", "Acl", nodeId, ALL_CLUSTERS_CTRL_TEST_ENDPOINT_ID)
+    # assert err == 0
+    # assert res.value[0] == entry0
+    # assert res.value[1] == entry1
+    # assert res.value[2] == entry2
+    # assert res.value[3] == entry3
```
Comment (pan-) [inline src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py:127]: Do we need that commented out code ?

File: src/test_driver/openiotsdk/integration-tests/shell/test_app.py [hunk 4]
```diff
@@ -95,6 +95,9 @@ def test_command_check(device):
     ret = device.wait_for_output("Open IoT SDK shell example application run")
     assert ret != None and len(ret) > 0
 
+    # Wait for printing prompt ">", we can't wait for it because it is not terminated by a newline character
+    sleep(0.5)
```
Comment (pan-) [inline src/test_driver/openiotsdk/integration-tests/shell/test_app.py:99]: I suggest increase of the timeout.

File: src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py (hunk 3 above)
Comment (ATmobica) [reply src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py:127]: Of course not. It was only for testing. Restored

**Comment by pan-** [comment]
Looks like it timeout during ota:

```
2023-02-13T12:41:49.9771497Z 12:41:49.976   INFO   <--|FVPdev| [INF] [SWU] New version of the software is available: 2
2023-02-13T12:41:49.9782950Z 12:41:49.977   INFO   New software image downloading and installing...
2023-02-13T13:01:50.1343410Z FAILED                                                                   [100%]
```

**Comment by ATmobica** [comment]
> Looks like it timeout during ota:
> 
> ```
> 2023-02-13T12:41:49.9771497Z 12:41:49.976   INFO   <--|FVPdev| [INF] [SWU] New version of the software is available: 2
> 2023-02-13T12:41:49.9782950Z 12:41:49.977   INFO   New software image downloading and installing...
> 2023-02-13T13:01:50.1343410Z FAILED                                                                   [100%]
> ```

It's the issue with that configuration that required debugging.

**Pull request merged**